cirq/linalg/decompositions.py

```diff
@@ -854,3 +857,129 @@ def kak_decomposition(unitary_object: Union[np.ndarray, 'cirq.SupportsUnitary'],
         global_phase=g * inner_cannon.global_phase,
         single_qubit_operations_before=(b1, b0),
         single_qubit_operations_after=(a1, a0))
+
+
+def kak_vector(unitary: Union[Iterable[np.ndarray], np.ndarray],
+               atol: float = 1e-9) -> np.ndarray:
+    r"""Compute the KAK vectors of one or more two qubit unitaries.
+
+    Any 2 qubit unitary may be expressed as
+
+    $$ U = k_l A k_r $$
+    where $k_l, k_r$ are single qubit (local) unitaries and
+    $$A= \exp\left(i\sum_{s=x,y,z} k_s \sigma_{s}^{(0)} \sigma{s}^{(0)}\right)$$
+
+    The vector entries are ordered such that
+        $$ 0 ≤ abs(k_z) ≤ k_y ≤ k_x ≤ π/4 $$
+            if $k_x$ = π/4, $k_z \geq 0$
+
+    Args:
+        unitary: A unitary matrix, or a multi-dimensional array of unitary
+            matrices. Must have shape (..., 4, 4), where the last two axes are
+            for the unitary matrix and other axes are for broadcasting the kak
+            vector computation.
+        atol: How close $k_x$ must be to π/4 to guarantee $k_z$ >= 0. Must be
+            positive.
+
+    Returns:
+        The KAK vector of the given unitary or unitaries. The output shape is
+        the same as the input shape, except the two unitary matrix axes are
+        replaced by the kak vector axis (i.e. the output has shape
+        `unitary.shape[:-2] + (3,)`).
+
+    References:
+        The appendix section of "Lower bounds on the complexity of simulating
+        quantum gates".
+        http://arxiv.org/abs/quant-ph/0307190v1
+
+    Examples:
+        >>> cirq.kak_vector(np.eye(4))
+        array([0., 0., 0.])
+        >>> unitaries = [cirq.unitary(cirq.CZ),cirq.unitary(cirq.ISWAP)]
+        >>> cirq.kak_vector(unitaries) * 4/np.pi
+        array([[ 1.,  0., -0.],
+               [ 1.,  1.,  0.]])
+    """
+    unitary = np.asarray(unitary)
+
+    if unitary.ndim < 2 or unitary.shape[-2:] != (4, 4):
+        raise ValueError(f'Expected input unitary to have shape (...,4,4), but'
+                         'got {unitary.shape}.')
```

Comment:
Does this actually format? There's no `f` on the second string.

Reply:
Good catch.

Reply:
Done.